client/menus.js

```diff
@@ -192,4 +192,149 @@ exports.menuitems = [
       window.openUILinkIn("https://addons.mozilla.org/firefox/collections/mozilla/webdeveloper/", "tab");
     }
   },
+  {
+    id: "dev-edition-profile",
+    l10nKey: "devEditionProfile",
+    oncommand(event) {
+      let profileName = "dev-edition";
+      let { Cc, Ci, Cu } = require("chrome");
+      // Create dev-edition profile and get path to it
+      let profileService = Cc["@mozilla.org/toolkit/profile-service;1"]
+        .getService(Ci.nsIToolkitProfileService);
+
+      let profile;
+      try {
+        // getProfileByName throws if it doesn't exists yet
+        profile = profileService.getProfileByName(profileName);
+      } catch(e) {}
+      if (!profile) {
+        profile = profileService.createProfile(null, profileName);
+        profileService.flush();
+      }
+      let profilePath = profile.rootDir.path;
+
+      let firefox_bin = Services.dirsvc.get("XREExeF", Ci.nsIFile);
+
+      if (Services.appinfo.OS == "WINNT") {
+        let linkName = "Mozilla Developer Edition.lnk";
+
+        // Create shortcut file to current firefox binary
+        // and with -profile command line argument refering to dev-edition profile
+        let shortcut = Services.dirsvc.get("TmpD", Ci.nsIFile);
+        shortcut.append(linkName);
+
+        // Previous failure may have let some junk
+        if (shortcut.exists()) {
+          try {
+            shortcut.remove(false);
+          } catch(e) {}
+        }
+        shortcut.createUnique(Ci.nsIFile.NORMAL_FILE_TYPE, 0o666);
+
+        /* function nsILocalFileWin.setShortcut(targetFile, workingDir, args,
+                                                description, iconFile, iconIndex) */
+        shortcut.QueryInterface(Ci.nsILocalFileWin);
+        shortcut.setShortcut(firefox_bin,
+          firefox_bin.parent,
+          "-profile \""+profilePath.replace("\\", "\\\\")+"\" -no-remote",
+          "Mozilla Developer Edition",
+          firefox_bin,
+          0);
+
+        // Copy the shortcut to Desktop and Start menu
+        let desktop = Services.dirsvc.get("Desk", Ci.nsIFile);
+        let progs = Services.dirsvc.get("Progs", Ci.nsIFile);
+        shortcut.copyTo(desktop, shortcut.leafName);
+        shortcut.copyTo(progs, shortcut.leafName);
+
+        // Remove the temporary shortcut file
+        shortcut.remove(false);
+      } else if (Services.appinfo.OS == "Darwin") {
```

Comment:
There might be a simpler approach for macOS, such as a shell script using `open -a <APP> --args <PROFILE>`, and some of them might get around the signing issues. Also, I think by default `/Applications` won't be writable by the current user. You could avoid this by writing to `~/Applications`, which is a standard per-user app directory.

There is a (somewhat old) [SO discussion][1] about ways to do this.

[1]: http://superuser.com/questions/16750/how-can-i-run-an-application-with-command-line-arguments-in-mac-os